### PR TITLE
Implement Batch injector missing events

### PR DESCRIPTION
**Pull request by marcopaulocova**
Added - Support for negative values in the rules
Added - Batch Before, Batch After and Block End Events
Updated - Tests for settings_view and validation_rule_enforcer
Updated - Batch Injector Factory
Updated - Block Info injector return statements for missing events
Updated - Block Validator
Moved - Block Info into a dedicated Injectors folder

Co-authored-by: AntonioBranco <antonio.branco@voidsoftware.com> and  Micaelsf <micael.ferreira@voidsoftware.com>

### CODE REVIEW

File: validator/src/journal/candidate_block.rs [hunk 1]
```diff
@@ -39,6 +39,27 @@ use pylogger;
 
 use scheduler::Scheduler;
 
+#[derive(PartialEq, Debug)]
+pub enum BatchInjectionStage {
+    BlockStart,
+    BeforeBatch,
+    AfterBatch,
+    BlockEnd,
+    BlockUnderFormation,
```
Comment (arsulegai) [inline validator/src/journal/candidate_block.rs:48]: Could you please add a comment on how is this used?
Is this a default not-set value?

**Comment by vaporos** [comment]
Thanks for the patience - please re-open this PR against the 1-3 branch. Once we get it merged there, we will backport it to 1-2.

**Comment by AntonioBrancoVoidSoftware** [comment]
There might be an issue with that. We saw your latest changes and the files that we changed were moved to the sawtooth-lib repo.
Do you still want us to re-open it?

**Comment by vaporos** [comment]
> There might be an issue with that. We saw your latest changes and the files that we changed were moved to the sawtooth-lib repo.
> Do you still want us to re-open it?

The 1-3 branch is prior to the libsawtooth move (which was done in master -- and master is now 1.99). The 1-3 branch becomes the "next 1.x release" branch, so we want to make sure changes land there first so we don't have regressions in features should we release a 1.3. Then we will backport to 1-2.

**Comment by agunde406** [comment]
Should the changes that were merged into 1-3 still be backported to the 1-2 branch? @marcopaulocova

**Comment by vaporos** [comment]
If we still need these changes backported to 1-2, please open a backport PR (or re-open this PR with that intent). Thanks!

**Pull request closed without merging**